tempodb/backend/cache/cache.go

```diff
@@ -110,6 +110,11 @@ func (r *readerWriter) CloseAppend(ctx context.Context, tracker backend.AppendTr
 	return r.nextWriter.CloseAppend(ctx, tracker)
 }
 
+func (r *readerWriter) Delete(ctx context.Context, name string, keypath backend.KeyPath) error {
+	// TODO also clear from cache
```

Comment:
we should do this before merging. otherwise it's a difficult to find bug in the future

Reply:
This comment is misleading, it's intended to say "once we implement caching, we will have to delete stuff from the cache here". We don't have caching yet so I'm just going to delete this comment.

Reply:
> To allow merging this PR maybe I can panic instead?

I suppose there's no code path to this b/c you don't implement caching on the overrides api? i'm fine with a panic. heads up @zalegrala

Reply:
Maybe log a message and return nil?  I suppose panic might be warranted if we really never intent to add support.

Reply:
I have changed it to return a panic for now. This isn't as clean as returning nil/logging an error but it would ensure we don't accidentally use this in production. If someone isn't careful while testing they might never call delete through a cache or not check the logs of their app. Having the application crash should ensure it doesn't fly under the radar.